lib/stripe/subscriptions/subscription.ex

```diff
@@ -145,10 +145,23 @@ defmodule Stripe.Subscription do
   Takes the `id` and an optional map of `params`.
   """
   @spec delete(Stripe.id() | t, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
-  def delete(id, opts \\ []) do
+  def delete(id, opts) when is_list(opts), do: delete(id, %{}, opts)
+
+  @spec delete(Stripe.id() | t, params) :: {:ok, t} | {:error, Stripe.Error.t()}
+        when params: %{
+               optional(:at_period_end) => boolean
+             }
+  def delete(id, params) when is_map(params), do: delete(id, params, [])
```

Comment:
Technically we wouldn't need the third `[]` since it defaults to that...

Reply:
Maybe I'm missing something, but wouldn't that make that clause call itself infinitely?

Reply:
I wish I could delete that comment :)